eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -147,7 +148,13 @@ public function copySubtree(APILocation $subtree, APILocation $targetParentLocat
         }
 
         // check create permission on target
-        if (!$this->permissionResolver->canUser('content', 'create', $loadedSubtree->getContentInfo(), [$loadedTargetLocation])) {
+        $locationTarget = (new DestinationLocationTarget($targetParentLocation->id, $loadedSubtree->contentInfo));
+        if (!$this->permissionResolver->canUser(
+            'content',
+            'create',
+            $loadedSubtree->getContentInfo(),
+            [$loadedTargetLocation, $locationTarget]
+        )) {
             throw new UnauthorizedException('content', 'create', ['locationId' => $loadedTargetLocation->id]);
         }
 
```

Comment:
We're modifying here lines which contain duplicated code fragment, adding more duplication, which makes Sonar gate fail. Lines 151-183 and 729-761 could be extracted into a single common method:
```php
    private function checkCreatePermissionOnSubtreeTarget(
        APILocation $targetParentLocation,
        APILocation $loadedSubtree,
        APILocation $loadedTargetLocation
    ): void {
        $locationTarget = (new DestinationLocationTarget(
            $targetParentLocation->id,
            $loadedSubtree->contentInfo
        ));
        if (!$this->permissionResolver->canUser(
            'content',
            'create',
            $loadedSubtree->getContentInfo(),
            [$loadedTargetLocation, $locationTarget]
        )) {
            throw new UnauthorizedException(
                'content',
                'create',
                ['locationId' => $loadedTargetLocation->id]
            );
        }

        // Check read access to whole source subtree
        $contentReadCriterion = $this->permissionCriterionResolver->getPermissionsCriterion(
            'content',
            'read'
        );
        if ($contentReadCriterion === false) {
            throw new UnauthorizedException('content', 'read');
        } elseif ($contentReadCriterion !== true) {
            // Query if there are any content in subtree current user don't have access to
            $query = new Query(
                [
                    'limit' => 0,
                    'filter' => new CriterionLogicalAnd(
                        [
                            new CriterionSubtree($loadedSubtree->pathString),
                            new CriterionLogicalNot($contentReadCriterion),
                        ]
                    ),
                ]
            );
            $result = $this->repository->getSearchService()->findContent($query, [], false);
            if ($result->totalCount > 0) {
                throw new UnauthorizedException('content', 'read');
            }
        }
    }
```